src/WingetCreateCLI/Commands/UpdateCommand.cs

```diff
@@ -199,43 +200,44 @@ public async Task<Manifests> DeserializeExistingManifestsAndUpdate(List<string>
                 return null;
             }
 
-            if (!PackageParser.UpdateInstallerNodesAsync(
-                installerManifest,
-                this.InstallerUrls,
-                packageFiles,
-                out bool installerMismatch,
-                out List<WingetCreateCore.Models.Installer.Installer> unmatchedInstallers,
-                out List<WingetCreateCore.Models.Installer.Installer> multipleMatchedInstallers,
-                out List<PackageParser.DetectedArch> detectedArchOfInstallers))
-            {
-                DisplayMismatchedArchitectures(detectedArchOfInstallers);
+            List<PackageParser.DetectedArch> detectedArchOfInstallers;
 
-                if (installerMismatch)
+            try
+            {
+                if (!PackageParser.UpdateInstallerNodesAsync(
+                    installerManifest,
+                    this.InstallerUrls,
+                    packageFiles,
+                    out detectedArchOfInstallers))
                 {
-                    Logger.ErrorLocalized(nameof(Resources.NewInstallerUrlMustMatchExisting_Message));
+                    DisplayMismatchedArchitectures(detectedArchOfInstallers);
+                    Logger.ErrorLocalized(nameof(Resources.PackageParsing_Error));

```

Comment:
This still looks like some errrors are return via exceptions, and others via exit code.  Why do we have split error handling?

Reply:
I believe in this case that our errors should be handled with exceptions. That way we can obtain better information about the specific packages that failed parsing. I modified the method so that errors are handled consistently.